Interaction - Missions: CoP Missions 3-1 to 3-3

claywar opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

* Converts CoP Mission 3-1 ~ 3-3 to Interaction Framework
* Adds additional dialogue for both paths, including reminder messages
* Implements Timer, Zone, and Mount checks for Windurst Path
* Implements OnZoneOut luautils function, and adds to Interaction Framework
* No longer reset missionStatus variable for CoP missions
* Adds xi.player.onPlayerMount
* Adds importantEvent() to Interaction Framework (First priority Progress, then reverts to Event Priority)

Note: I touched Cid and Ru'Lude Gardens Zone, so CI is going to fail.

claywar committed:
Interaction - Mission: The Call of the Wyrmking

claywar committed:
Interaction - Mission: A Vessel Without a Captain

cocosolos committed:
Check in CoP MissionStatus documentation

claywar committed:
Interaction - Mission: The Road Forks

* Implements San d'Oria Path

Implement remaining items in PM3-3

claywar committed:
Implement OnZoneOut, do not reset missionStatus for CoP mission complete

zach2good commented on scripts/globals/effects/mounted.lua:
```diff
@@ -17,6 +17,15 @@ effect_object.onEffectGain = function(target, effect)
+
+    -- For PM3-3 The Road Forks.  This value will be checked periodically, and break
+    -- the Mimeo Jewel should a player be mounted, zone, or disconnect.
+    if
```
(Optional) This might be a good opportunity to put in `xi.player.onPlayerMounted` and move this check in to there

claywar replied:
I wonder, do we want to keep this in a global scope, or implement it in a specific zone?

claywar replied:
Done

zach2good commented on scripts/missions/cop/3_3_The_Road_Forks.lua:
```diff
@@ -0,0 +1,465 @@
+        local minutesRemaining = secondsRemaining / 60
+
+        if minutesRemaining <= nextMessageTime then
+            player:messageSpecial(attohwaChasmID.text.MIMEO_JEWEL_OFFSET + messageOffset - 1) -- TODO: Might need KI Arg
```
You could store this on the player as a localvar?

claywar replied:
messageOffset and Timer are both stored as a localVar.  It changes on each message update, though we're not 0-indexed for the messages displayed for safety (there's a lot of duplicates for the first message text, so its 1-indexed, with the first message being 1 lower).

zach2good commented on sql/npc_list.sql:
```diff
@@ -456,10 +456,9 @@ INSERT INTO `npc_list` VALUES (16806387,'HomePoint#5','Home Point #5',0,0.000,0.
--- NC: INSERT INTO `npc_list` VALUES (16806391,'qm','???',0,0.000,0.000,0.000,0,50,50,0,0,0,0,0,0x0000320000000000000000000000000000000000,0,NULL,0);
--- NC: INSERT INTO `npc_list` VALUES (16806392,'blank','     ',0,0.000,0.000,0.000,0,50,50,0,0,0,0,0,0x0000320000000000000000000000000000000000,0,NULL,0);
--- NC: INSERT INTO `npc_list` VALUES (16806393,'blank','     ',0,0.000,0.000,0.000,0,50,50,0,0,0,0,0,0x0000320000000000000000000000000000000000,0,NULL,0);
--- NC: INSERT INTO `npc_list` VALUES (16806394,'Harvesting_Point','Harvesting Point',0,0.000,0.000,0.000,0,50,50,0,0,0,0,0,0x0000320000000000000000000000000000000000,0,NULL,0);
```
Did we lose `16806394`?

claywar replied:
We did.  Does not appear in March update when looking at the mob lists

zach2good commented on src/map/zone.cpp:
```diff
@@ -1008,6 +1008,8 @@ void CZone::CharZoneOut(CCharEntity* PChar)
         }
     }
 
+    luautils::OnZoneOut(PChar);
```
👍

claywar committed:
QC: CoP Missions 3-1 through 3-3


Update Timer function to return if no KI present for all cases


Add additional dialogue for San d'Oria Path


Fix Syntax Error


Add additional dialogue for first Yoran interaction


Remove unnecessary afterZoneIn check, add additional dialogue and update Loose Sand range


Additional dialogue


Additional dialogue and cleanup


Fix bad condition for Overgrown Ivy


Additional cleanup, implement importantEvent modifier


Parameter, syntax, and action fixes

claywar committed:
Implement onPlayerMount

claywar commented:
Need to re-audit NPC sql changes.  Will push an update tomorrow which also includes xi.player.onPlayerMount

claywar committed:
Add NPC animation for Mimeo Jewel, Add missing NPCs

claywar commented:
Tested completely, should be good to go.
